clu/src/lib.rs

```diff
@@ -92,21 +86,65 @@ fn linux_init(config: Arc<RwLock<settings::RitaSettingsStruct>>) -> Result<(), E
     cleanup()?;
     KI.restore_default_route(&mut config.get_network_mut().default_route)?;
 
-    let privkey = config.get_network().wg_private_key.clone();
-    let pubkey = config.get_network().wg_public_key.clone();
-    let mesh_ip = config.get_network().own_ip.clone();
+    let mut network_settings = config.get_network_mut();
+    let privkey = network_settings.wg_private_key.clone();
+    let pubkey = network_settings.wg_public_key.clone();
+    let mesh_ip_option = network_settings.mesh_ip.clone();
+    let own_ip_option = network_settings.own_ip.clone(); // TODO: REMOVE IN ALPHA 11
+
+    match mesh_ip_option {
+        Some(existing_mesh_ip) => {
+            if !validate_mesh_ip(&existing_mesh_ip) {
+                warn!(
+                    "Existing mesh_ip field {} is invalid, generating a new mesh IP",
+                    existing_mesh_ip
+                );
+                network_settings.mesh_ip =
+                    Some(linux_generate_mesh_ip().expect("failed to generate a new mesh IP"));
+            } else {
+                info!("Mesh IP is {}", existing_mesh_ip);
+            }
+        }
 
-    if !validate_wg_key(&privkey) || !validate_wg_key(&pubkey) {
-        linux_generate_wg_keys(&mut config.get_network_mut()).expect("failed to generate wg keys");
+        // Fall back and migrate from own_ip if possible, TODO: REMOVE IN ALPHA 11
+        None => match own_ip_option {
+            Some(existing_own_ip) => if validate_mesh_ip(&existing_own_ip) {
+                info!(
+                    "Found existing compat own_ip field {}, migrating to mesh_ip",
+                    existing_own_ip
+                );
+                network_settings.mesh_ip = Some(existing_own_ip);
+            } else {
+                warn!(
+                    "Existing compat own_ip value {} is invalid, generating a new mesh IP and migrating to mesh_ip",
+                    existing_own_ip
+                    );
+                network_settings.mesh_ip =
+                    Some(linux_generate_mesh_ip().expect("failed to generate a new mesh IP"));
```

Comment:
Wouldn't it make sense here to propagate possible error here (and in similiar places) instead of a panic?

Reply:
This is the only place where the IP is generated, and without said IP there's no point running the application.